deps/oculus/include/ovrsession.h

```diff
@@ -0,0 +1,188 @@
+#ifndef _OVR_SESSION_H_
+#define _OVR_SESSION_H_
+
+#include <nan.h>
+#include <v8.h>
+#include <webgl.h>
+
+#include "OVR_CAPI_GL.h"
+
+typedef struct SwapChain {
+  ovrTextureSwapChain ColorTextureChain;
+  ovrTextureSwapChain DepthTextureChain;
+  ovrSizei textureSize;
+} EyeSwapChain;
+
+namespace oculusvr {
+  class OculusVRPosRes;
+  void RunResInMainThread(uv_async_t *handle);
+}
+
+class OculusVRPosRes {
+public:
+  OculusVRPosRes(Local<Function> cb);
+  ~OculusVRPosRes();
+
+  Nan::Persistent<Function> cb;
+};
+
+class OVRSession : public Nan::ObjectWrap
+{
+public:
+  static NAN_MODULE_INIT(Init);
+
+  // Static factory construction method for other node addons to use.
+  static v8::Local<v8::Object> NewInstance();
+
+private:
+  explicit OVRSession();
+  ~OVRSession() = default;
+
+  // Node construction method for new instances.
+  static NAN_METHOD(New);
+  static NAN_METHOD(GetControllersInputState);
+  static NAN_METHOD(GetPose);
+  static NAN_METHOD(Submit);
+  static NAN_METHOD(GetRecommendedRenderTargetSize);
+
+  /// Create a singleton reference to a constructor function.
+  static inline Nan::Persistent<v8::Function>& constructor()
```

Comment:
This is more a suggestion, but I would keep implementations in the c++ side of this file, for compilation performance and code duplication reasons.

If we include this file from two objects, the linker might reject it as an error, which would be surprising and confusing.

Reply:
I've split declaration / definition